### DISCUSSION TITLE
switch inside form is triggering submit

### DISCUSSION THREAD

**Discussion by ramseylove** [react-hook-form/react-hook-form#10107] [Q&A]
HI all,

I have a form with a switch element from a internal component libary, which is built from a button internally. I am using the controller component to control it via react hook form. it looks like this:

```<Controller
        name="active"
        control={Form.control}
        render={({ field }) => (
          <SwitchWithLabels
            ref={field.ref}
            label='Active'
            labelPosition="left"
            fieldLabelWidth={'160px'}
            isOn={field.value}
            onPress={(newState) => field.onChange(newState)}
          />
        )}
      />
```

I trigger the submit remotely via another component. 
`Form.handleSubmit(onSubmit)()`

All this works fine in another component but when i added this switch and trigger it, the form is submitted. This is obviously not the way it should work and i'm not sure what possibly to add to stop react hook form from acting like the switch is a submit button. Im a little limited by the component library i am using but all suggestions are welcomed.

**Comment by leapful** [comment]
@ramseylove 

You can try to inspect real DOM button element of component `SwitchWithLabels` to see if attribute `type` of button has been set or not.

By default, `<button></button>` without attribute `type` will be a submit button equivalent to `<button type="submit"></button>` that will cause a submit event on your form.

If so, you need find the DOM button element (maybe by ID or class or any identifier of the component `SwitchWithLabels`) and set attribute `type="button` to solve this problem.

**Reply by ramseylove** [reply]
It looks like the type attribute is not forwarded  on to the underling button component, which is the problem. I was able to dig into the component library code and add the type attr, and it fixed the issue.

I hoping to figure out a way to programmatically add it to the dom element as a better work around then editing the library code. Not sure that its possible though

Thanks for the help @leapful

**Reply by ramseylove** [reply]
I had access to the ref, so i added the attribute in a useEffect hook 
```
const switchRef = useRef<HTMLButtonElement | null>(null);
  useEffect(() => {
    switchRef.current?.setAttribute('type', 'button');
  }, [switchRef]);
```
